src/model/Layer.ts

```diff
@@ -59,37 +59,45 @@ export interface SearchConfig {
 }
 
 export interface DefaultLayerClientConfig {
-  minResolution?: number;
-  maxResolution?: number;
-  hoverable?: boolean;
-  searchable?: boolean;
-  searchConfig?: SearchConfig;
-  propertyConfig?: DefaultLayerPropertyConfig[];
-  featureInfoFormConfig?: PropertyFormTabConfig<PropertyFormItemReadConfig>[];
-  editFormConfig?: PropertyFormTabConfig<PropertyFormItemEditDefaultConfig |
-    PropertyFormItemEditReferenceTableConfig>[];
   crossOrigin?: string;
-  opacity?: number;
   downloadConfig?: DownloadConfig[];
+  editFormConfig?: PropertyFormTabConfig<PropertyFormItemEditDefaultConfig |
+    PropertyFormItemEditReferenceTableConfig>[];
   editable?: boolean;
+  featureInfoFormConfig?: PropertyFormTabConfig<PropertyFormItemReadConfig>[];
+  hoverable?: boolean;
+  maxResolution?: number;
+  minResolution?: number;
+  opacity?: number;
+  propertyConfig?: DefaultLayerPropertyConfig[];
+  searchConfig?: SearchConfig;
+  searchable?: boolean;
 }
 
-export interface LayerArgs extends BaseEntityArgs {
+export type DefaultLayer = Layer<DefaultLayerClientConfig, DefaultLayerSourceConfig>;
```

Comment:
Is this type really needed? We could also pass the `DefaultLayerClientConfig` and `DefaultLayerSourceConfig` types as defaults to the existing `Layer`.

This would probably make this PR non breaking?

Reply:
I thought it would be a bit more readable and intuitive. Do the changes from 52ac3e082a2629d16f350426b95f84bdc413d689 correspond to what you had in mind?

Reply:
I think it can simplified even more by providing default types for the generic argument, quite similar to the [Application](https://github.com/terrestris/shogun-util/blob/main/src/model/Application.ts#L83-L88) model. Not tested, but it might look as follows:

```ts
export default class Layer<
  D extends DefaultLayerClientConfig = DefaultLayerClientConfig,
  S extends DefaultLayerSourceConfig = DefaultLayerSourceConfig
> extends BaseEntity {
  (…)
}
```

This way it won't be necessary to actually specify the generic types if no custom type should be used.